code/game/objects/effects/abnormality.dm

```diff
@@ -37,3 +37,77 @@
 /obj/effect/qoh_sygil/proc/fade_out()
 	animate(src, alpha = 0, time = 1 SECONDS)
 	QDEL_IN(src, 1 SECONDS)
+
+/obj/effect/magic_bullet
+	name = "magic bullet"
+	desc = "A black bullet wreathed in blue-white energy, screaming forth at an unfathomable speed."
+	icon = 'icons/obj/projectiles.dmi'
+	icon_state = "magicbullet"
+	throwforce = 100
+	move_force = INFINITY
+	move_resist = INFINITY
+	pull_force = INFINITY
+	density = TRUE
+	anchored = TRUE
+	generic_canpass = FALSE
+	movement_type = PHASING | FLYING
+	var/already_moved = 0
+
+/obj/effect/magic_bullet/Initialize()
+	..()
+	playsound(get_turf(src), 'sound/abnormalities/freischutz/shoot.ogg', 100, 1)
+
+/obj/effect/magic_bullet/Moved() // Shamelessly stolen code from immovable rod and paradise lost, GO!
+	var/list/nearmiss = list()
+	if(dir == (EAST||WEST))
+		nearmiss += get_step(src, SOUTH).contents
+		nearmiss += get_step(src, NORTH).contents
+	else
+		nearmiss += get_step(src, EAST).contents
+		nearmiss += get_step(src, WEST).contents
+	for(var/mob/living/L in loc.contents)
+		L.apply_damage(120, BLACK_DAMAGE, null, L.run_armor_check(null, BLACK_DAMAGE), spread_damage = TRUE)
+		visible_message("<span class='boldwarning'>[src] pierces through [L]!</span>")
+		to_chat(L, "<span class='userdanger'>[src] slams through you!</span>")
+	for(var/mob/living/L in nearmiss)
+		L.apply_damage(60, BLACK_DAMAGE, null, L.run_armor_check(null, BLACK_DAMAGE), spread_damage = TRUE)
+		visible_message("<span class='boldwarning'>[src] just barely brushes past [L]!</span>")
+		to_chat(L, "<span class='userdanger'>[src] grazes your side!</span>")
+	if((src.x > 245) || (src.x < 10) || (src.y > 245) || (src.y < 10))
```

Comment:
This would cause issues when fired from the left edge of the map.

Reply:
All of our maps are centered at roughly (128, 128) last I checked, and none of them come even close to being 128 tiles in any direction from the center. If this becomes a problem I can just make it delete itself if it detects it's left the Z-level, but letting it onto another Z-level for even a second might have even more disastrous consequences down the line, so for now I want to err towards safety.